test/extended/apiserver/tls.go

```diff
@@ -1,31 +1,146 @@
 package apiserver
 
 import (
+	"context"
 	"crypto/tls"
+	"fmt"
+	"strings"
+	"time"
 
 	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/openshift/library-go/test/library"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned"
 	"github.com/openshift/library-go/pkg/crypto"
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
+const tlsTestDuration = 45 * time.Minute
+const tlsWaitForCleanupDuration = 10 * time.Minute
+
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Serial][Slow]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("apiserver")
+
+	g.It("TestTLSModernProfile", func() {
+		ctx, ctxCancelFn := context.WithTimeout(context.Background(), tlsTestDuration)
+		defer ctxCancelFn()
+
+		t := g.GinkgoT()
+
+		configClient := oc.AdminConfigClient()
+
+		operatorClient := oc.AdminOperatorClient()
+
+		defer func() {
```

Comment:
maybe we should use `DeferCleanup` - i think that it will be executed even if the test panics ...